### PR TITLE
ci: update snapshots that fail with requests resource name change

### PR DESCRIPTION
**Pull request by ZStriker19**
[This change](https://github.com/DataDog/dd-trace-py/pull/7717) broke ci since langchain and openai rely on the requests library, but the tests were not run ([fixed here](https://github.com/DataDog/dd-trace-py/pull/7776)).
This fixes the snapshots that fail, and also deletes the starlette snapshots that would fail if they were run (they're no longer run since they're for python 3.6).

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: tests/snapshots/tests.contrib.openai.test_openai_v0.test_integration_service_name[None-None].json [hunk 1]
```diff
@@ -8,12 +8,11 @@
     "parent_id": 0,
     "meta": {
       "_dd.p.dm": "-0",
-      "_dd.p.tid": "654a694400000000",
+      "_dd.p.tid": "65674d7100000000",
       "component": "openai",
       "language": "python",
       "openai.api_base": "https://api.openai.com/v1",
       "openai.api_type": "open_ai",
-      "openai.base_url": "https://api.openai.com/v1/",
```
Comment (ZStriker19) [inline tests/snapshots/tests.contrib.openai.test_openai_v0.test_integration_service_name[None-None].json:16]: Why is base_url getting deleted with no replacement here? Is it related to the resource name change or something else?

Comment (ZStriker19) [reply tests/snapshots/tests.contrib.openai.test_openai_v0.test_integration_service_name[None-None].json:16]: I noticed that in some cases `base_url` is missing, and not replaced, and in some cases it’s `api_base` or `api_type` .  Looking at the code though, and the Openai versions for which we don’t have certain tags, these seem to be correct…. e.g.[ we’re missing api_base and api_type here using >openai==1.0.0](https://github.com/DataDog/dd-trace-py/pull/7778/files#diff-bca930a237a23fc479cbd7a4a881a0afc62599f18a2bb23890e43a699969c613) which makes sense given the code here: https://github.com/DataDog/dd-trace-py/blob/2.x/ddtrace/contrib/openai/patch.py#L185-L197
So… why didn’t we catch this earlier?

Comment (mabdinur) [reply tests/snapshots/tests.contrib.openai.test_openai_v0.test_integration_service_name[None-None].json:16]: @Yun-Kim

Comment (Yun-Kim) [reply tests/snapshots/tests.contrib.openai.test_openai_v0.test_integration_service_name[None-None].json:16]: Discussed offline, I overlooked this when creating snapshots for v0/v1 - we reuse the same snapshots for most tests for v0/v1 (which is why we added `base_url/api_type/api_base` tags to all snapshots) but not for `test_integration_service_name` 🤦 so we can go ahead with these changes, thanks!

**Review by emmettbutler — APPROVED** [review]

**Comment by emmettbutler** [comment]
@ZStriker19 some openai tests are still failing

**Pull request merged**